src/java/htsjdk/samtools/sra/SRAAlignmentIterator.java

```diff
@@ -0,0 +1,198 @@
+/*===========================================================================
+*
+*                            PUBLIC DOMAIN NOTICE
+*               National Center for Biotechnology Information
+*
+*  This software/database is a "United States Government Work" under the
+*  terms of the United States Copyright Act.  It was written as part of
+*  the author's official duties as a United States Government employee and
+*  thus cannot be copyrighted.  This software/database is freely available
+*  to the public for use. The National Library of Medicine and the U.S.
+*  Government have not placed any restriction on its use or reproduction.
+*
+*  Although all reasonable efforts have been taken to ensure the accuracy
+*  and reliability of the software and data, the NLM and the U.S.
+*  Government do not and cannot warrant the performance or results that
+*  may be obtained by using this software or data. The NLM and the U.S.
+*  Government disclaim all warranties, express or implied, including
+*  warranties of performance, merchantability or fitness for any particular
+*  purpose.
+*
+*  Please cite the author in any work or product based on this material.
+*
+* ===========================================================================
+*
+*/
+
+package htsjdk.samtools.sra;
+
+import htsjdk.samtools.*;
+import ngs.*;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Created by andrii.nikitiuk on 9/3/15.
+ */
+public class SRAAlignmentIterator implements Iterator<SAMRecord> {
+    private ValidationStringency validationStringency;
+
+    private ReadCollection run;
+    private SAMFileHeader header;
+    private final List<Reference> cachedReferences;
+    private List<Long> referencesLengths;
+    private Iterator<Chunk> referencesChunksIterator;
+    private int currentReference = -1;
+
+    private boolean hasMoreReferences = true;
+
+    private AlignmentIterator alignedIterator;
+    private Boolean hasMoreAlignments = false;
+
+    private SRALazyRecord lastRecord;
+
+    public static List<Long> getReferencesLengths(ReadCollection run) throws ErrorMsg {
+        ReferenceIterator refIt = run.getReferences();
+        List<Long> lengths = new ArrayList<Long>();
+        while (refIt.nextReference()) {
+            long refLen = refIt.getLength();
+            // lets optimize references so they always align in 5000 bases positions
+            if (refLen % 5000 != 0) {
+                refLen += 5000 - (refLen % 5000);
```

Comment:
Also add a comment explaining why are you aligning to multiples of 5000 rather than reporting the true reference lengths.